modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -34,25 +35,89 @@ PolygonMeshGeneratorBase::generate()
+    const multiBdryLayerParams & duct_outer_boundary_layer_params,
+    bool has_rings,
+    bool has_ducts,
+    const Real pitch,
+    const unsigned int num_sectors_per_side,
+    const unsigned int background_intervals,
+    const Real background_radial_bias,
+    const singleBdryLayerParams & background_inner_boundary_layer_params,
+    const singleBdryLayerParams & background_outer_boundary_layer_params,
+    dof_id_type & node_id_background_meta,
+    const unsigned int side_number,
+    const unsigned int side_index,
+    const std::vector<Real> azimuthal_tangent,
+    const subdomain_id_type block_id_shift,
+    const bool quad_center_elements,
+    const boundary_id_type boundary_id_shift)
 {
   auto mesh = buildReplicatedMesh(2);
 
+  // Calculate biasing terms
+  // background region needs to be split into three parts
+  const auto main_background_bias_terms =
+      biasTermsCalculator(background_radial_bias, background_intervals);
+  const auto inner_background_bias_terms =
+      biasTermsCalculator(background_inner_boundary_layer_params.bias,
+                          background_inner_boundary_layer_params.intervals);
+  const auto outer_background_bias_terms =
+      biasTermsCalculator(background_outer_boundary_layer_params.bias,
+                          background_outer_boundary_layer_params.intervals);
+  auto rings_bias_terms = biasTermsCalculator(ring_radial_biases,
+                                              rings,
+                                              ring_inner_boundary_layer_params.fractions,
+                                              ring_inner_boundary_layer_params.intervals,
+                                              ring_inner_boundary_layer_params.biases,
+                                              ring_outer_boundary_layer_params.fractions,
+                                              ring_outer_boundary_layer_params.intervals,
+                                              ring_outer_boundary_layer_params.biases);
+  auto duct_bias_terms = biasTermsCalculator(duct_radial_biases,
+                                             ducts_layers,
+                                             duct_inner_boundary_layer_params.fractions,
+                                             duct_inner_boundary_layer_params.intervals,
+                                             duct_inner_boundary_layer_params.biases,
+                                             duct_outer_boundary_layer_params.fractions,
+                                             duct_outer_boundary_layer_params.intervals,
+                                             duct_outer_boundary_layer_params.biases);
+
+  std::vector<unsigned int> total_rings;
+  for (unsigned int i = 0; i < rings.size(); i++)
+    total_rings.push_back(rings[i] + ring_inner_boundary_layer_params.intervals[i] +
+                          ring_outer_boundary_layer_params.intervals[i]);
+
+  if (background_inner_boundary_layer_params.intervals)
```

Comment:
so you're tagging the inner background boundary layer onto the pin rings
and the outer background boundary layer onto the duct rings. 

Why is that?

Reply:
After discussing with TH people, it seems that boundary layer needs to have a uniform thickness for better TH code performance and analyses. Namely, they need to be conformal to the boundary shape. In this case, the inner boundary layer needs to be circular and the outer boundary layer needs to be "duct" shape. So the easiest approach is to mesh them as if they are rings/ducts.

Reply:
thanks for the explanation